recipe/meta.yaml

```diff
@@ -102,9 +105,10 @@ outputs:
         - scipy
         - anndata <0.10 # [py>37]
         - anndata <0.9  # [py<=37]
-        - tiledb-py >=0.24.0,<0.25.0
+        - tiledb-py >=0.25.0,<0.26.0
         - typing-extensions >=4.1
-        - numba
+        - numba >=0.58.1 # [py>37]
+        - numba          # [py<=37]
```

Comment:
I confirmed that this matches the upstream requirements in `setup.py`

https://github.com/single-cell-data/TileDB-SOMA/blob/5dac84866c5dfc74f0ff4f9b891c9b26902846c7/apis/python/setup.py#L283

Note that `~=0.58.0` is actually stricter than `>=`. The `~=` translates to a pin of `0.58.*`. However, I agree with the choice of `>=` for the conda recipe. It's better to allow more flexibility for the conda solver until we know for certain there is a limitation in the version range. We can always submit a repodata patch in the future to fix the metadata if a hypothetical numba 0.59 were to cause problems